### ISSUE TITLE
download from yt-dlp error

### ISSUE THREAD

**Issue by candrapersada** [axcore/tartube#317]
why download from yt-dlp error
```
Traceback (most recent call last):
  File "D:/tartube/mingw64/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "D:/tartube/mingw64/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "D:/tartube/mingw64/bin/yt-dlp.exe/__main__.py", line 4, in <module>
  File "D:/tartube/mingw64/lib/python3.8/site-packages/yt_dlp/__init__.py", line 16, in <module>
    from .options import (
  File "D:/tartube/mingw64/lib/python3.8/site-packages/yt_dlp/options.py", line 8, in <module>
    from .compat import (
  File "D:/tartube/mingw64/lib/python3.8/site-packages/yt_dlp/compat.py", line 3, in <module>
    import asyncio
ModuleNotFoundError: No module named 'asyncio'

```
MS windows
Tartube Portable v2.3.286
yt-dlp 2021.08.02

**Comment by MohamedEssabani** [comment]
@candrapersada How did you get that error output? I think I may have the same error when downloading with yt-dlp, but nothing is showing in the "Output" tab of tartube.

**Comment by axcore** [comment]
The author of yt-dlp has added a new dependency, and that doesn't work with the old version of MSYS2.

I have to update MSYS2 to the latest version, which fixes your problem.

**Comment by candrapersada** [comment]
what are the yt-dlp dependencies?

**Comment by axcore** [comment]
asyncio. You need the latest version of MSYS2. When https://github.com/axcore/tartube/issues/316 is resolved, I will do a new release.

**Comment by ceonelson** [comment]
I spent a few hours a couple weeks ago figuring out how to fix this but forgot to post it. Here's the solution:

I updated yt-dlp and can no longer download new videos because of asyncio error.
found this thread: https://cygwin.com/pipermail/cygwin/2021-February/247681.html
sure enough, my python was 3.8.7:
C:\Users\Owner\Documents\___Apps\Tartube\msys64\mingw64\bin\python3.8.exe --version
Python 3.8.7

Traceback (most recent call last):
  File "C:\users\owner\documents\___apps\tartube\msys64\mingw64\lib\python3.8\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\users\owner\documents\___apps\tartube\msys64\mingw64\lib\python3.8\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\Owner\Documents\___Apps\Tartube\msys64\mingw64\bin\yt-dlp.exe\__main__.py", line 4, in <module>
  File "C:\users\owner\documents\___apps\tartube\msys64\mingw64\lib\python3.8\site-packages\yt_dlp\__init__.py", line 16, in <module>
    from .options import (
  File "C:\users\owner\documents\___apps\tartube\msys64\mingw64\lib\python3.8\site-packages\yt_dlp\options.py", line 8, in <module>
    from .compat import (
  File "C:\users\owner\documents\___apps\tartube\msys64\mingw64\lib\python3.8\site-packages\yt_dlp\compat.py", line 3033, in <module>
    import asyncio
ModuleNotFoundError: No module named 'asyncio'

enter into the wingw64 terminal: tartube\msys64\mingw64.exe
$ pacman -Su
$ pacman -Sy mingw-w64-x86_64-python3

You should be all set!

**Comment by ceonelson** [comment]
Also what I did until I figured out the workaround was just download an older release of yt-dlp and point tartube to it. I think 2021-06-09 will work, but if not the 2021-04-11 release definitely does.

**Comment by axcore** [comment]
Fixed in v2.3.306. There is a new installer for MS Windows.
